### PR TITLE
Variables doc updates

### PR DESCRIPTION
**Pull request by jakekaplan**
Update docs to reflect new python sdk and cli methods

### CODE REVIEW

File: docs/guides/variables.md [hunk 1]
```diff
@@ -57,38 +62,40 @@ You can access any variable through the Python SDK with the `Variable.get()` met
...
+answer = Variable.get("not_the_answer", default="42")
+print(answer) # 42
+
+# overwrite an existing variable
+Variable.set(name="the_answer", value="43", overwrite=True)
+answer = Variable.get("the_answer")
+print(answer) #43
 ```
 
+All standard methods are also available as async methods:
```
Comment (zangell44) [inline docs/guides/variables.md:89]: When should I use sync methods vs. async methods?

File: docs/guides/variables.md [hunk 2]
```diff
@@ -57,38 +62,40 @@ You can access any variable through the Python SDK with the `Variable.get()` met
...
-
-# using `overwrite=True`
-answer = Variable.get('the_answer')
-print(answer.value)
-#42
-answer = Variable.set(name="the_answer", value="43", overwrite=True)
-print(answer.value)
-#43
+# set a variable
+Variable.set(name="the_answer", value="42")
```
Comment (zangell44) [inline docs/guides/variables.md:66]: Any reason not to keep the assignment in here?
```suggestion
variable = Variable.set(name="the_answer", value="42")
```

Comment (jakekaplan) [reply docs/guides/variables.md:66]: We changed the interface so `set` doesn't return anything but now I am reconsidering it. `get` returns the value by default and the full object optionally, going to get a couple more opinions on what "feels good" here

File: docs/guides/variables.md (hunk 1 above)
Comment (jakekaplan) [reply docs/guides/variables.md:89]: good call, need to update this because sync version won't work outside of tasks/flows anymore

**Comment by jakekaplan** [comment]
closing this in favor of a different pr

**Pull request closed without merging**